tasks/adjust_config.yml

```diff
@@ -33,3 +33,71 @@
+  name: Setup preview generator widthSizes
+  ansible.builtin.command:
+    cmd: |-
+      docker exec --user={{ nextcloud_uid }}:{{ nextcloud_gid }} {{ nextcloud_identifier }}-server php /var/www/html/occ config:app:set previewgenerator widthSizes  --value={{ nextcloud_preview_widthSizes }}
+
+- when: nextcloud_preview_enabled | bool
+  name: Setup preview generator heightSizes
+  ansible.builtin.command:
+    cmd: |-
+      docker exec --user={{ nextcloud_uid }}:{{ nextcloud_gid }} {{ nextcloud_identifier }}-server php /var/www/html/occ config:app:set previewgenerator heightSizes  --value={{ nextcloud_preview_heightSizes }}
+
+- when: nextcloud_preview_enabled | bool
+  name: Setup preview generator preview_max_x
+  ansible.builtin.command:
+    cmd: |-
+      docker exec --user={{ nextcloud_uid }}:{{ nextcloud_gid }} {{ nextcloud_identifier }}-server php /var/www/html/occ config:system:set preview_max_x --value {{ nextcloud_preview_preview_max_x }}
+
+- when: nextcloud_preview_enabled | bool
+  name: Setup preview generator preview_max_y
+  ansible.builtin.command:
+    cmd: |-
+      docker exec --user={{ nextcloud_uid }}:{{ nextcloud_gid }} {{ nextcloud_identifier }}-server php /var/www/html/occ config:system:set preview_max_y --value {{ nextcloud_preview_preview_max_y }}
+
+- when: nextcloud_preview_enabled | bool
+  name: Setup preview generator config:system:set jpeg_quality
+  ansible.builtin.command:
+    cmd: |-
+      docker exec --user={{ nextcloud_uid }}:{{ nextcloud_gid }} {{ nextcloud_identifier }}-server php /var/www/html/occ config:system:set jpeg_quality --value {{ nextcloud_preview_system_jpeg_quality }}
+
+- when: nextcloud_preview_enabled | bool
+  name: Setup preview generator app:set preview jpeg_quality
+  ansible.builtin.command:
+    cmd: |-
+      docker exec --user={{ nextcloud_uid }}:{{ nextcloud_gid }} {{ nextcloud_identifier }}-server php /var/www/html/occ config:app:set preview jpeg_quality --value={{ nextcloud_preview_app_jpeg_quality }}
+
+- when: nextcloud_preview_enabled | bool
+  name: "Check if {{ nextcloud_preview_first_run_finished_filename }} exists"
+  ansible.builtin.stat:
+    path: "{{ nextcloud_preview_host_folder }}/{{ nextcloud_preview_first_run_finished_filename }}"
+  register: nextcloud_preview_file_stats
+
+- when: nextcloud_preview_enabled | bool and not nextcloud_preview_file_stats.stat.exists
+  name: Start preview generate-all
+  ansible.builtin.command:
+    cmd: |-
+      docker exec --user={{ nextcloud_uid }}:{{ nextcloud_gid }} {{ nextcloud_identifier }}-server php /var/www/html/occ preview:generate-all
+  async: 100000
+  poll: 10
+
+- when: nextcloud_preview_enabled | bool
+  name: Create file to signal, that first run is executed
+  ansible.builtin.file:
+    path: "{{ nextcloud_preview_host_folder }}/{{ nextcloud_preview_first_run_finished_filename }}"
+    state: "touch"
+    mode: 0644
+    owner: "{{ nextcloud_uid }}"
+    group: "{{ nextcloud_gid }}"
```

Comment:
Instead of having many tasks, each one with `when: ...`, you can have a `block` and nest them all inside.

Example:

```yaml
- when: nextcloud_preview_enabled | bool
  block:
    - name: Setup preview generator squareSizes
      .........

    - name: Setup preview generator widthSizes
      ........
```

I suppose you could also make use of Ansible's `with_items` and use a single task for many of these "set parameter" calls.

Reply:
Moved everything into a single block and also merged the occ config set tasks